### ISSUE TITLE
Get rid of forward primer in forward reverse reads

### ISSUE THREAD

**Issue by termithorbor** [marcelm/cutadapt#483]
Hi,

before primer removal I can find my primers like this:
```
                 Forward Complement Reverse RevComp
FWD.ForwardReads   88149          0       0       0
FWD.ReverseReads       9          0       0     113
REV.ForwardReads      11          0       0     305
REV.ReverseReads   74775          0       0       0
```
After running this code I end up with this:

FWD = forward primer sequence
REV = reverse primer sequence
FWD.RC <- dada2:::rc(FWD)
REV.RC <- dada2:::rc(REV)

R1.flags <- paste0("-a ", FWD,"...", REV.RC) 

R2.flags <- paste0("-A ", REV, "...", FWD.RC) 

for(i in seq_along(fnFs)) {
  system2(cutadapt, args = c(R1.flags, R2.flags, 
                             "-o", fnFs.cut[i], "-p", fnRs.cut[i], 
                             fnFs.filtN[i], fnRs.filtN[i])) 
}
```
                 Forward Complement Reverse RevComp
FWD.ForwardReads       0          0       0      17
FWD.ReverseReads     107          0       0       0
REV.ForwardReads     104          0       0       0
REV.ReverseReads       0          0       0      14
```
So basically I am not able to get rid of forward primer sequences in the forward reverse reads and in the reverse forward reads and the opposite with the reverse primer.

How do I need to modify my code to get rid of those and are they even a problem for downstream analysis with tools like DADA2 or Qiime2? 
It is just 16S V3-4 amplicon data, so probably it is just by chance a similar sequence somewhere in the part of the gene in some microbes and not actual primer sequence?

Many thanks in advance.

**Comment by marcelm** [comment]
What are the numbers shown in the tables?

**Comment by termithorbor** [comment]
1. number = primer hits in forward sequence; 2. number = primer hits in the forward complement; 3. number = primer hits in reverse sequence; 4. number = primer hits in reverse complement

1. table = primer hits before cutadapt
2. table = primer hits after cutadapt

**Comment by marcelm** [comment]
I’m having some trouble understanding what you mean. For example, you write “forward reverse reads” and “reverse forward reads”, which I don’t know how to place.

I also don’t understand why some of the numbers in the second table should be higher than the corresponding ones in the first.

I’ll just explain a little bit and perhaps that helps. Sorry if something that I say is already obvious to you.

You should concentrate on the forward sequence and its reverse complement since these are the two options that occur physically (because that is how polymerases work).

It is true that sometimes people describe a sequence as “reversed”, but that is just sloppy terminology. They actually mean “reverse-complemented”. The same goes for “forward” (R1) and “reverse” (R2) reads: The second read is reverse-complemented relative to the first one.

So you should ignore the second and third column of your tables (they are all zero anyway, precisely because these operations don’t correspond to anything that would happen on the molecular level).

Cutadapt does not know about reverse complements and it will not reverse-complement anything (neither adapters nor reads), so all the adapters that you search for must be in the same orientation as the reads on which you search for them. (This changes with `--revcomp`, but that option is not supported for paired-end reads at the moment anyway).

It looks like your command line includes `-a FWD...REV.RC` and `-A REV...FWD.RC`, which appears to be correct for your use case.

My advice is create a minimal working example (MWE), that is, to find a read pair that is not trimmed the way you would expect it. Then compare the input and the output; perhaps you can figure out what is going on. Feel also free to post it here, perhaps it’s a peculiarity of the adapter finding algorithm in Cutadapt.

**Comment by termithorbor** [comment]
Hi,                     

I will add up to this topic now.

![image](https://user-images.githubusercontent.com/66436023/104741193-2a5e8a00-5749-11eb-987f-82661ddaa65d.png)

So when I search for my primer somehow the forward primer is found in the reverse reads and in the reverse complement of the forward reads and those are not removed by my cutadapt command -a FWD...REV.RC and -A REV...FWD.RC:

![image](https://user-images.githubusercontent.com/66436023/104741425-790c2400-5749-11eb-8a9b-70ce07cda92b.png)

Why are my primers found there? is it just biological variance? So just true biological sequences which are similar to my primers but not actually the primers?

Thanks in advance.

**Comment by marcelm** [comment]
I haven’t looked at this problem in particular before, but from what I hear, these things are usually not biological sequences, but library preparation artifacts. I would never expect 100% clean results and I actually think that getting only 0.1% weird reads looks quite good to me.

**Comment by marcelm** [comment]
I’m closing this issue now, hoping I was able to help. If you have more questions or think the issue should be re-opened, please let me know.
